Azure-Samples/angular-cosmosdb
# CSRF Protection

Please see #5 first before reviewing/merging this PR

## Purpose

* Adds protection against CSRF attacks with the csurf module and Angular

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
    * Run the app with browser dev tools network tab open
    * Login
    * In a new tab, go to [https://www.clarkio.com/csrf-attack](https://www.clarkio.com/csrf-attack)
    * Click the "Click to execute the attack" button
    * Confirm you see an error message: "ForbiddenError: invalid csrf token"


## What to Check
Verify that the following are valid
* You receive an error when attempting to execute the CSRF attack against the application

## Other Information
